src/StocksDisplay.jsx

```diff
@@ -6,34 +6,42 @@ import AddRow from './AddRow.jsx';
 
 
 class StocksDisplay extends Component {
+    views = [] ;
 
-  render() {
-    let existingCategory = '' ; 
-    let views = [] ;
-    this.props.stocks.map( stock => {
-        
-        if( ( this.props.showInStock && !stock.stocked ) || ( stock.name.indexOf(this.props.searchText) === -1 ) )
-            return ;
-
-        if( stock.category !== existingCategory){
-            views.push( < AddCategory title = {stock.category} />)
-        }
-
-        views.push( < AddRow  content = {stock} />)
-        existingCategory = stock.category ;
-
-    });
-    return (
-        <div className = "App" >
-            <Row>
-                <Col xs='6' className = "nameHeading"> NAME </Col>  
-                <Col xs ='6' className = " priceHeading "> PRICE </Col>  
-           </Row>
-           <hr />
-           {views}
-        </div>
-    );
-  }
+    componentWillReceiveProps= ( nextProps ) => { 
+        let existingCategory = '' ; 
+        let newStock = nextProps.stocks[0];
+        newStock.map( stock => {
+            
+            if( ( nextProps.showInStock && !stock.stocked ) || ( stock.name.indexOf(nextProps.searchText) === -1 ) )
+                return ;
+
+            if( stock.category !== existingCategory){
+                this.views.push( < AddCategory title = {stock.category} />)
+            }
+
+            this.views.push( < AddRow  content = {stock} />)
+            existingCategory = stock.category ;
+
```

Comment:
This logic should not be here. We should always try to separate logic from the view layer.
Have a look at presentational vs container components here http://redux.js.org/docs/basics/UsageWithReact.html